ui/components/app/snaps/tx-insight-warnings/tx-insight-warnings.js

```diff
@@ -0,0 +1,199 @@
+  }, {});
+
+  const [warningState, setWarningState] = useState(defaultWarningState);
+  const [isChecked, setIsChecked] = useState(false);
+
+  const handleOnChange = () => setIsChecked((state) => !state);
+
+  const handleWarningClick = (snapId) => {
+    let newState = { ...warningState };
+    const nextSnapState = !warningState[snapId];
+    const willClose = nextSnapState;
+    newState[snapId] = nextSnapState;
+    if (!willClose) {
+      newState = { ...defaultWarningState };
+      newState[snapId] = false;
+      newState[warnings[0].snapId] = snapId !== warnings[0].snapId;
+    }
+    setWarningState(newState);
+  };
+
+  const Warnings = () => {
+    const lastWarningIdx = warnings.length - 1;
+    return (
+      <Box className="tx-insights-warnings-modal__content">
+        {warnings.map((warning, idx) => {
+          const { snapId, content } = warning;
+          return (
+            <SnapUIRenderer
+              key={`${snapId}-${idx}`}
+              snapId={snapId}
+              data={content}
+              delineatorType={DelineatorType.Warning}
+              onClick={() => handleWarningClick(snapId)}
+              isCollapsable
+              isCollapsed={warningState[snapId]}
+              boxProps={{ marginBottom: idx === lastWarningIdx ? 0 : 4 }}
+            />
+          );
+        })}
+      </Box>
+    );
+  };
+
+  // move this to an enum that defines the language to be used in this modal on the various
+  // screens that will offer transaction insights, should be indexed on "type".
+  const results = {
+    confirming: { noun: 'confirmation', imperative: 'confirm' },
+    signing: { noun: 'signature', imperative: 'sign' },
+  };
+
+  return (
+    <Modal
+      isOpen
+      isClosedOnEscapeKey={false}
+      isClosedOnOutsideClick={false}
+      className="tx-insights-warnings-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          childrenWrapperProps={{
+            alignItems: AlignItems.center,
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+          }}
+        >
+          <AvatarIcon
+            iconName={IconName.Warning}
+            size={AvatarIconSize.Lg}
+            color={IconColor.errorDefault}
+            backgroundColor={BackgroundColor.errorMuted}
+          />
+          <Text
+            variant={TextVariant.headingMd}
+            paddingTop={4}
+            paddingBottom={4}
+          >
+            {t('transactionInsightWarningHeader')}
+          </Text>
+        </ModalHeader>
+        <Text variant={TextVariant.bodyMd} paddingBottom={4}>
+          {warnings.length > 1
+            ? t('transactionInsightWarningContentPlural', [
+                warnings.length,
+                type,
+                results[type].noun,
+              ])
+            : t('transactionInsightWarningContentSingular', [
+                type,
+                results[type].noun,
+              ])}
```

Comment:
0 should use the plural form too ("0 warnings"). It's probably easiest to invert the condition:

```ts
warnings.length === 1 ? t('singular') : t('plural')
```